lib/hotel_admin.rb

```diff
@@ -0,0 +1,134 @@
+require 'date'
+require 'date_range'
+require_relative 'room'
+require_relative 'reservation'
+require_relative 'BookingSystem_Errors'
+
+module BookingSystem
+  class HotelAdmin
+
+    attr_reader :room_list, :reservation_list, :block_list
+
+    def initialize
+      #hotel admin knows all reservations (non-block and block)
+      @reservation_list = []
+      @block_list = []
+      #hotel admin knows all rooms in hotel
+      @room_list = BookingSystem::Room.all
+    end
+
+    def reserve_room(first_name, last_name, room_id, room_rate, start_date, end_date, block_id = nil)
+
+      raise InvalidDateRangeError.new("Date range conflicts with room requested") if end_date <= start_date
+      requested_range = DateRange.new(start_date, end_date)
+
+      if block_id.nil? #if a room doesn't have a block_id, blacklist all rooms that are blocked
+        raise UnavailableRoomError.new("Room is unavailable") if @block_list.any? do |block|
+          block.date_range.overlaps?(requested_range) && block.rooms_array.any? { |room| room.id == room_id }
```

Comment:
This couples `Block (and similarly `Reservation`) very tightly.  Instead you could create a method in Reservation & Block that checks to see if it contains the room/date combo.